### PR TITLE
Paginate query results for (hopefully) softer DB queries

### PR DESCRIPTION
**Pull request by Willdotwhite**

### CODE REVIEW

File: app/src/main/kotlin/team/finder/api/teams/TeamsService.kt [hunk 1]
```diff
@@ -10,8 +12,8 @@ class TeamsService(val repository: TeamsRepository) {
 
     fun createTeam(team: Team) = repository.save(team)
 
-    fun getTeams(skillsetMask: Int?): List<Team> {
-        var records = repository.findAll().filter { team -> team?.deletedAt.equals(null) }
+    fun getTeams(pageable: Pageable, skillsetMask: Int?): List<Team> {
+        var records: Streamable<Team> = repository.findAll(pageable).filter { team -> team?.deletedAt.equals(null) }
```
Comment (dallonf) [inline app/src/main/kotlin/team/finder/api/teams/TeamsService.kt:16]: This and the below is doing DB-side filtering, right? If not, does it work with pagination?

Comment (Willdotwhite) [reply app/src/main/kotlin/team/finder/api/teams/TeamsService.kt]: Ahh, you were too early! I've offloaded a lot of this to the `@Query` directly, but yes - the pagination occurs _after_ the query WHERE clauses. 

(It didn't previously though, good spot)

File: app/src/main/kotlin/team/finder/api/teams/TeamsController.kt [hunk 2]
```diff
@@ -9,7 +11,18 @@ import javax.validation.Valid
...
 
     @GetMapping("/teams")
-    fun index(@RequestParam skillsetMask: Int?) : List<Team> = service.getTeams(skillsetMask)
+    fun index(@RequestParam(defaultValue = "1") page: Int, @RequestParam skillsetMask: Int?) : List<Team> {
+
+        // Pagination needs to be offset by -1 from expectations, but can't be set below 0
+        val pageIdx = if (page > 0) page else 1
+        val queryPageable: PageRequest = PageRequest.of(pageIdx - 1, 50)
+
+        return if (skillsetMask?.equals(null) == false) {
```
Comment (Boivin92) [inline app/src/main/kotlin/team/finder/api/teams/TeamsController.kt:20]: technically if I ask for a skillsetMask of 0 you'd be doing the masking query for nothing

**Review by Boivin92 — APPROVED** [review]

File: app/src/main/kotlin/team/finder/api/teams/TeamsController.kt (hunk 2 above)
Comment (Willdotwhite) [reply app/src/main/kotlin/team/finder/api/teams/TeamsController.kt:20]: Oh yeah, that should be limited to a minimum of 1 as well. A mask of 0 is meaningless.

Wish I'd noticed that before turning my computer off 🤦

**Review by dallonf — APPROVED** [review]

**Pull request merged**